### ISSUE TITLE
HyperV: Hung at Downloading kubeadm v1.10.0

### ISSUE THREAD

**Issue by borysl** [kubernetes/minikube#2977]
**Is this a BUG REPORT or FEATURE REQUEST?** (choose one):
BUG REPORT

Please provide the following details:

**Environment**:

**Minikube version** (use `minikube version`):  v0.28.0
- **OS** (e.g. from /etc/os-release): Windows 10 (17134.165) UA
- **VM Driver** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep DriverName`): hyperv
- **ISO version** (e.g. `cat ~/.minikube/machines/minikube/config.json | grep -i ISO` or `minikube ssh cat /etc/VERSION`): "Boot2DockerURL": "file://C:/Users/MSI/.minikube/cache/iso/minikube-v0.28.0.iso"
- **Install tools**: bash version GNU bash, version 4.3.46(2)-release (x86_64-pc-msys)
- **Others**: Run under admin mode, docker is already up and running

**What happened**:
Trying to start minikube on hyperv on Windows 10 via 
$ minikube start 
**What you expected to happen**:
Started minikube and set running context for kubectl
**How to reproduce it** (as minimally and precisely as possible):
Mostly followed this article: https://medium.com/@JockDaRock/minikube-on-windows-10-with-hyper-v-6ef0f4dc158c
```
$ minikube start --vm-driver hyperv --hyperv-virtual-switch "Primary Virtual Switch"
Starting local Kubernetes v1.10.0 cluster...
Starting VM...
Getting VM IP address...
Moving files into cluster...
Downloading kubelet v1.10.0
Downloading kubeadm v1.10.0
```
 then the command line just hangs
(when it should have been continued with 
```
Finished Downloading kubelet v1.10.0
Finished Downloading kubeadm v1.10.0
...
```
)
also tried many other ways including attempt to install earlier version:
$ minikube start --kubernetes-version="v1.9.4"
Starting local Kubernetes v1.9.4 cluster...
Starting VM...
Getting VM IP address...
Kubernetes version downgrade is not supported. Using version: v1.10.0
Moving files into cluster...
Downloading kubeadm v1.10.0
Downloading kubelet v1.10.0

**Output of `minikube logs` (if applicable)**:
See above
**Anything else do we need to know**:
The machine itself is started on hyperv, however I cannot shut it down or use, I can open terminal, but I don't know which login and password to use, or how to access via ssh right now.

**Comment by CarlosLanderas** [comment]
I am suffering the same problem.

**Comment by seenaomi** [comment]
I'm experiencing the exact same thing on win10/virtualbox

**Comment by seenaomi** [comment]
This helped me: https://github.com/kubernetes/minikube/issues/2765

**Comment by dautzenb** [comment]
I had the same issue on Windows 10/Hyper-V. I installed v0.25.0 and it worked again.

**Comment by willemodendaal** [comment]
I had a similar issue on v0.30 on Windows 10/Hyper-V and wifi. Happened a couple of times, then I switched to ethernet instead and it got past those download steps.

**Comment by ilkka** [comment]
I had the same problem just now with v0.32, but managed to work around it by not creating an external virtual switch but an internal one, then sharing my wifi connection with it. Found instructions here: https://niftysoft.github.io/devlog/2018/06/30/hyperv-and-minikube-windows-10-without-external.html

**Comment by tstromberg** [comment]
If someone runs into this again, do you mind including debug log output? 

`minikube start -v=9`

I'm really curious what it's blocked on: is it the download, or is it the file transfer to the VM. Thanks!

**Comment by yami12376** [comment]
I have the same issue on `v0.33.1` minikube.
Hangs on `Downloading kubelet v1.13.2`

**Comment by yami12376** [comment]
I've figured it out myself - on Windows 10  Hyper-V should be turn on if you are using Docker desktop but for minikube it should be turn off.

When i had turned off Hyper-V and restarted my PC - then` minikube start` worked fine.

**Comment by TomaszLacinaGL** [comment]
Hello, I have same problem

my logs are 

```
 minikube start -v=9 --vm-driver hyperv --hyperv-virtual-switch "Primary Virtual Switch" --logtostderr
W0313 07:36:21.169563   16604 root.go:146] Error reading config file at C:\Users\tomasz.lacina\.minikube\config\config.json: open C:\Users\tomasz.lacina\.minikube\config\config.json: The system cannot find the file specified.
I0313 07:36:21.172578   16604 start.go:102] Viper configuration:
Aliases:
map[string]string{}
Override:
map[string]interface {}{"v":"9"}
PFlags:
map[string]viper.FlagValue{"gpu":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fb360)}, "network-plugin":viper.pflagValue{flag:(*pflag.Flag)(0xc0002faf00)}, "service-cluster-ip-range":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fab40)}, "profile":viper.pflagValue{flag:(*pflag.Flag)(0xc0004ad220)}, "docker-env":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa780)}, "docker-opt":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa820)}, "apiserver-names":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa960)}, "host-only-cidr":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa3c0)}, "vm-driver":viper.pflagValue{flag:(*pflag.Flag)(0xc0004adf40)}, "hyperv-virtual-switch":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa460)}, "mount":viper.pflagValue{flag:(*pflag.Flag)(0xc0004adcc0)}, "uuid":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fb180)}, "extra-config":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fb0e0)}, "memory":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa0a0)}, "apiserver-ips":viper.pflagValue{flag:(*pflag.Flag)(0xc0002faa00)}, "apiserver-name":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa8c0)}, "cache-images":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fb040)}, "kubernetes-version":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fae60)}, "xhyve-disk-driver":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa5a0)}, "disk-size":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa320)}, "dns-domain":viper.pflagValue{flag:(*pflag.Flag)(0xc0002faaa0)}, "iso-url":viper.pflagValue{flag:(*pflag.Flag)(0xc0004adea0)}, "keep-context":viper.pflagValue{flag:(*pflag.Flag)(0xc0004adc20)}, "kvm-network":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa500)}, "cpus":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa1e0)}, "cri-socket":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fadc0)}, "disable-driver-mounts":viper.pflagValue{flag:(*pflag.Flag)(0xc0004ade00)}, "registry-mirror":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fac80)}, "container-runtime":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fad20)}, "mount-string":viper.pflagValue{flag:(*pflag.Flag)(0xc0004add60)}, "nfs-shares-root":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa6e0)}, "hyperkit-vsock-ports":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fb2c0)}, "insecure-registry":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fabe0)}, "nfs-share":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fa640)}, "bootstrapper":viper.pflagValue{flag:(*pflag.Flag)(0xc0004ad2c0)}, "feature-gates":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fafa0)}, "hyperkit-vpnkit-sock":viper.pflagValue{flag:(*pflag.Flag)(0xc0002fb220)}}
Env:
map[string]string{}
Key/Value Store:
map[string]interface {}{}
Config:
map[string]interface {}{}
Defaults:
map[string]interface {}{"wantupdatenotification":true, "wantkubectldownloadmsg":true, "wantnonedriverwarning":true, "showbootstrapperdeprecationnotification":true, "v":"0", "alsologtostderr":"false", "reminderwaitperiodinhours":24, "wantreporterror":false, "wantreporterrorprompt":true, "showdriverdeprecationnotification":true, "log_dir":""}
Starting local Kubernetes v1.12.4 cluster...
Starting VM...
I0313 07:36:21.199562   16604 cluster.go:69] Machine does not exist... provisioning new machine
I0313 07:36:21.204562   16604 cluster.go:70] Provisioning machine with config: {MinikubeISO:https://storage.googleapis.com/minikube/iso/minikube-v0.32.0.iso Memory:2048 CPUs:2 DiskSize:20000 VMDriver:hyperv ContainerRuntime: HyperkitVpnKitSock: HyperkitVSockPorts:[] XhyveDiskDriver:ahci-hd DockerEnv:[] InsecureRegistry:[] RegistryMirror:[] HostOnlyCIDR:192.168.99.1/24 HypervVirtualSwitch:Primary Virtual Switch KvmNetwork:default Downloader:{} DockerOpt:[] DisableDriverMounts:false NFSShare:[] NFSSharesRoot:/nfsshares UUID: GPU:false}
I0313 07:36:21.204562   16604 downloader.go:56] Not caching ISO, using https://storage.googleapis.com/minikube/iso/minikube-v0.32.0.iso
Reading certificate data from C:\Users\tomasz.lacina\.minikube\certs\ca.pem
Decoding PEM data...
Parsing certificate...
Reading certificate data from C:\Users\tomasz.lacina\.minikube\certs\cert.pem
Decoding PEM data...
Parsing certificate...
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive @(Get-Module -ListAvailable hyper-v).Name | Get-Unique
[stdout =====>] : Hyper-V

[stderr =====>] :
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive @([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole("S-1-5-3
2-578")
[stdout =====>] :
[stderr =====>] : Exception calling "IsInRole" with "1" argument(s): "The trust relationship between the primary domain and the trusted domain failed.
"
At line:1 char:1
+ @([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIde ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [], MethodInvocationException
    + FullyQualifiedErrorId : SystemException


exit status 1
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive @([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")
[stdout =====>] : True

[stderr =====>] :
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (Hyper-V\Get-VMSwitch).Name
[stdout =====>] : DockerNAT
Primary Virtual Switch
Default Switch

[stderr =====>] :
Downloading C:\Users\tomasz.lacina\.minikube\cache\boot2docker.iso from file://C:/Users/tomasz.lacina/.minikube/cache/iso/minikube-v0.32.0.iso...
Creating SSH key...
Creating VM...
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive (Hyper-V\Get-VMSwitch).Name
[stdout =====>] : DockerNAT
Primary Virtual Switch
Default Switch

[stderr =====>] :
Using switch "Primary Virtual Switch"
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive @([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent()).IsInRole([Security.Principal.WindowsBuiltInRole] "Administrator")
[stdout =====>] : True

[stderr =====>] :
Creating VHD
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\New-VHD -Path 'C:\Users\tomasz.lacina\.minikube\machines\minikube\fixed.vhd' -SizeBytes 10MB -Fixed
[stdout =====>] :

ComputerName            : WRO1-LDL-P12122
Path                    : C:\Users\tomasz.lacina\.minikube\machines\minikube\fixed.vhd
VhdFormat               : VHD
VhdType                 : Fixed
FileSize                : 10486272
Size                    : 10485760
MinimumSize             :
LogicalSectorSize       : 512
PhysicalSectorSize      : 512
BlockSize               : 0
ParentPath              :
DiskIdentifier          : 7BDB6A39-EC0D-4A55-A0FD-65E128AAE951
FragmentationPercentage : 0
Alignment               : 1
Attached                : False
DiskNumber              :
IsPMEMCompatible        : False
AddressAbstractionType  : None
Number                  :




[stderr =====>] :
Writing magic tar header
Writing SSH key tar header
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\Convert-VHD -Path 'C:\Users\tomasz.lacina\.minikube\machines\minikube\fixed.vhd' -DestinationPath 'C:\Users\tomasz.lacina\.minikube\machines\minikube\disk.vhd' -VHDType Dynamic -DeleteSource
[stdout =====>] :
[stderr =====>] :
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\Resize-VHD -Path 'C:\Users\tomasz.lacina\.minikube\machines\minikube\disk.vhd' -SizeBytes 20000MB
[stdout =====>] :
[stderr =====>] :
[executing ==>] : C:\Windows\System32\WindowsPowerShell\v1.0\powershell.exe -NoProfile -NonInteractive Hyper-V\New-VM minikube -Path 'C:\Users\tomasz.lacina\.minikube\machines\minikube' -SwitchName 'Primary Virtual Switch' -MemoryStartupBytes 2048MB
```

**Comment by borysl** [comment]
After several experiments during the last several month. I found the most easy way to work it out on Windows 10:
Just forget about external Virtual Switch and use internal.

1. Turned on the hosted network supported (use netsh wlan show drivers under admin to check if you have it already). 
https://www.youtube.com/watch?v=AiHF9f9xfMQ

2. Run hosted network mode (use NETSH WLAN start hostednetwork)
https://www.windowscentral.com/how-turn-your-windows-10-pc-wireless-hotspot

3. Run with internal switch nat:
https://niftysoft.github.io/devlog/2018/06/30/hyperv-and-minikube-windows-10-without-external.html

**Comment by nikitinalexander** [comment]
> I had the same problem just now with v0.32, but managed to work around it by not creating an external virtual switch but an internal one, then sharing my wifi connection with it. Found instructions here: https://niftysoft.github.io/devlog/2018/06/30/hyperv-and-minikube-windows-10-without-external.html

This helped me solve the issue.

minikube version: v1.4.0
commit: 7969c25a98a018b94ea87d949350f3271e9d64b6

Installed via Choco.

**Comment by mkdewangan** [comment]
I had the same problem. I created internal switch as per -  https://niftysoft.github.io/devlog/2018/06/30/hyperv-and-minikube-windows-10-without-external.html
My internal switch name -- minikube internal v switch
Internal Switch got appeared here in Control Panel\Network and Internet\Network Connections. In my computer there is "Network Bridge" icon is also here. which is showing connected. I think my computer using network bridge. I right clicked on Internal Switch(which we created) and clicked join the bridge. Then i ran below command , it worked . it downloaded kubeadm kubelet successfully. 

minikube start --vm-driver hyperv --hyperv-virtual-switch "minikube internal v switch"

Once this ran and command prompt return. I ran below command 

 kubectl get pods -n kube-system

It worked.
